### ISSUE TITLE
An option `purge_unused_elements_by_class` to purge all classes in bulk

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#4139]
Would it be possible to have an option `purge_unused_elements_by_class` to purge all classes in bulk?  That is, purging all of the following, at one time, for example.

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/507113/f64ddbfc-30b9-4302-9671-04fc1e6efb20)

**Comment by Andrej730** [comment]
Sounds scary for me atm, I'm not sure if all these classes can be safely purged (e.g. in this info message it's filtering out IfcStyledItem, relationships, contexts).

**Comment by theoryshaw** [comment]
How will we know, if we don't try? :)

**Comment by Moult** [comment]
So it's definitely scary, and this falls under the developer tools in the debug panel I'd say (as opposed to purge unused types and materials which have a valid user-facing usecase). So I've moved the UI there. If you don't specify a class it attempts to auto purge everything.

I went through every single IFC class and whitelisted the "safe" ones to merge. I probably made a mistake. We'll find out the hard way :)
